x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java

```diff
@@ -131,6 +140,11 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getTransportVersion().onOrAfter(TransportVersions.V_8_16_0)) {
             out.writeLong(queryStartTimeNanos);
         }
+        if (supportPartialResults(out.getTransportVersion())) {
+            out.writeBoolean(allowPartialResults);
+        } else if (allowPartialResults) {
+            throw new IllegalArgumentException("allow_partial_result is not supported in this version");
```

Comment:
Is this what we want to do in this case? If I send a CCS request with allow partials set and one of my clusters is old and does not support this option, I think I'd rather prefer it to just work without allowing partials than fail immediately. I mean, this whole thing is aimed at allowing queries to work in presence of some problems, so I think it makes sense to be more lenient here?

Reply:
The general principle in Elasticsearch is to reject queries that cannot be properly handled. Users expect queries with `allow_partial_results=false` to skip failures (e.g., missing shards) and return partial results. However, this won't be fulfilled if remote clusters or nodes in a mixed cluster are running older versions. Therefore, we should reject these queries. Users need to upgrade to leverage this feature.

Reply:
Wait, I'm not sure I got this right. In this case we're talking about `allow_partial_results=true` (since it's under `if (allowPartialResults)`) and this says "skip missing shards". It is true that on older clusters it's not possible, but isn't total query failure even worse? I would prefer an option to have at least some failures skipped to never being able to use it at all until all clusters everywhere (which may not even be under my control in CCS case, could be entirely different setup by other team) upgrade. I'd be in a situation that if I query even one single old cluster, even one that never fails, I am unable to use `allow_partial_results=true` for any of my own clusters, even upgraded ones. I think this is not the optimal result.

Reply:
I understand your point, but this is how we introduce new features in Elasticsearch. We should reject requests with allow_partial_results=true that are explicitly set by users if the clusters are not ready to handle them properly.

However, I've allowed this serialization to [pass through](https://github.com/elastic/elasticsearch/pull/121942/commits/926382977478b820944207d5963b3986e66226b5) so we can switch the default for allow_partial_results to true in versions 8.19 and 9.1.